### ISSUE TITLE
SQL on FHIR view layer - POC implementation

### ISSUE THREAD

**Issue by johngrimes** [aehrc/pathling#1438]
This issue represents the work to implement and validate the WIP spec for the SQL on FHIR view layer.

The spec is here: https://github.com/FHIR/sql-on-fhir-v2/blob/master/fhir_analytics_spec.md

**Comment by johngrimes** [comment]
Tasks remaining:

- [ ] Refactor implementation to use array representations
- [ ] Get all view tests passing, and complete test suite
- [ ] Implement extract operation on top of views
- [ ] Implement aggregate operation on top of views

**Comment by johngrimes** [comment]
Hi @axelv,

We're pretty much functionally complete on this, but we are working out what parts of this it makes sense to release and maintain into the future.

We're looking at making release 8 somewhat SQL on FHIR and bulk data access-centric, adding the view runner capability and bulk data client and server features but removing operations like aggregate and extract.

Are you able to share any information about your interest in the release? Are you planning on using this, and if so, what are your requirements?

**Comment by axelv** [comment]
Thx @johngrimes.

TLDR: We want to flatten QuestionnaireResponses.

At [tiro.health](https://www.tiro.health/) we are building SDC tools that for clinicians in hospitals. The output of these tools are FHIR Questionnaires & FHIR QuestionnaireResponses. Internally we have our own SQL on FHIR pipeline to flatten the QR data.
But data engineers at some of the hospitals want the raw FHIR data and flatten the QR's themselves. In such a scenario we want to share the ViewDefinitions. Some of the hospitals are working on Azure Databricks and ask us how to setup the pipeline in their environment. Pathling seems like natural fit for this...

**Comment by johngrimes** [comment]
@axelv Thanks for that info, it sounds like a great fit. We will get on to releasing this functionality as soon as possible.
